lib/postgrex/extensions/numeric.ex

```diff
@@ -8,6 +8,9 @@ defmodule Postgrex.Extensions.Numeric do
       %Decimal{} = decimal ->
         data = unquote(__MODULE__).encode_numeric(decimal)
         [<<IO.iodata_length(data)::int32>> | data]
+      n when is_float(n) ->
+        data = unquote(__MODULE__).encode_numeric(Decimal.from_float(n))
+        [<<IO.iodata_length(data)::int32>> | data]
       n when is_number(n) ->
```

Comment:
Let's change this check now to `is_integer`. `is_number` is basically `is_integer + is_float` and we covered the first part already.